opswat-metadefender-core/operations.py

```diff
@@ -0,0 +1,261 @@
+            logger.error('There was an error while handling the request')
+            raise ConnectorError('There was an error while handling the request')
+        except Exception as e:
+            logger.error('{0}'.format(e))
+            raise ConnectorError('{0}'.format(e))
+
+    def login(self):
+        headers = {
+            "Content-Type": "application/json"
+        }
+        payload = {
+            "user": self.username,
+            "password": self.password
+        }
+        response = self.make_rest_call('/login', data=json.dumps(payload), headers=headers)
+        return response.get('session_id')
+
+    def logout(self):
+        self.make_rest_call('/logout')
+
+
+def check_payload(payload):
+    updated_payload = {k: v for k, v in payload.items() if v is not None and v != ''}
+    return updated_payload
+
+
+def handle_params(params):
+    value = str(params.get('value'))
+    input_type = params.get('input_type')
+    try:
+        if isinstance(value, bytes):
+            value = value.decode('utf-8')
+        if input_type == 'Attachment ID':
+            if not value.startswith('/api/3/attachments/'):
+                value = '/api/3/attachments/{0}'.format(value)
+            attachment_data = make_request(value, 'GET')
+            file_iri = attachment_data['file']['@id']
+            file_name = attachment_data['file']['filename']
+            logger.info('file id = {0}, file_name = {1}'.format(file_iri, file_name))
+            return file_iri
+        elif input_type == 'File IRI':
+            if value.startswith('/api/3/files/'):
+                return value
+            else:
+                raise ConnectorError('Invalid File IRI {0}'.format(value))
+    except Exception as err:
+        logger.info('handle_params(): Exception occurred {0}'.format(err))
+        raise ConnectorError('Requested resource could not be found with input type "{0}" and value "{1}"'.format
+                             (input_type, value.replace('/api/3/attachments/', '')))
+
+
+def submitFile(file_iri):
+    try:
+        file_path = join('/tmp', download_file_from_cyops(file_iri)['cyops_file_path'])
+        logger.info(file_path)
+        with open(file_path, 'rb') as attachment:
+            file_data = attachment.read()
+        if file_data:
+            files = {'file': file_data}
+            return files
+        raise ConnectorError('File size too large, submit file up to 32 MB')
+    except Exception as Err:
+        logger.error('Error in submitFile(): %s' % Err)
+        raise ConnectorError('Error in submitFile(): %s' % Err)
+
+
+def submit_file(config, params):
+    opswat = Opswat(config)
+    try:
+        submit_type = params.get('submit_type')
+        submit_mode = params.get('submit_mode')
+        if submit_type == 'File Download URL':
+            if submit_mode == 'Synchronous Mode':
+                other_attributes = params.get('other_attributes')
+                payload = {
```

Comment:
write a common function to generate payload, it can be used in all 4 conditions

Reply:
done